src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -2222,6 +2222,25 @@ namespace {
     EXPECT_EQ(0, countNonzero(activeColumns));
   }
 
+
+  TEST(SpatialPoolerTest, testSameOutputForSameInputNoLearningNoBoosting)
+  {
+    // This test verifies possible issue where SP with 0-boosting and learning
+    // disabled produces different (random) output, compared to the python SP
+    const UInt inputSize = 10;
+    const UInt nColumns = 20;
+    SpatialPooler sp;
+    sp.initialize({inputSize}, {nColumns});
+    sp.setBoostStrength(0);
+
+    vector<UInt> input = { 1, 1, 0, 0, 1, 1, 0, 0, 1, 1 };
+    vector<UInt> out1(nColumns, 0);
+    vector<UInt> out2(nColumns, 0);
+    sp.compute(input.data(), false, out1.data());
+    sp.compute(input.data(), false, out2.data());
+    EXPECT_EQ(out1, out2);
```

Comment:
I don't think this test validates the issue. For 2 reasons: 
- your commit  eba72d0  (this test) passed OK even without the suggested fix
- the fix is in python related `.i` file. The issue exhibits even in pure c++
- please see #1382 for discussion and example test that crashes correctly